sympy/physics/mechanics/functions.py

```diff
@@ -552,14 +552,20 @@ def _f_list_parser(fl, ref_frame):
     """Parses the provided forcelist composed of items
     of the form (obj, force).
     Returns a tuple containing:
-        vlist: The velocity (ang_vel for Frames, vel for Points) in
+        vel_list: The velocity (ang_vel for Frames, vel for Points) in
                 the provided reference frame.
-        flist: The forces.
+        f_list: The forces.
 
     Used internally in the KanesMethod and LagrangesMethod classes.
     """
     def flist_iter():
-        for obj, force in fl:
+        for pair in fl:
+            if not pair:
+                # TODO: Handle the case where there is an empty tuple in an
+                # otherwise non-empty forcing list (which is a very rare case)
```

Comment:
Oh, I see what you mean. If someone passes in something like `((), (P, force))`. I don't think that we need to support this. You can specify in the docstring of kanes_equations that the loads should be simply None. We could make it only support:

1) `None`
2) an iterable of point/frame, vector tuples

Supporting any kind of empty iterable is actually a little odd and redundant.

In the other thread I mentioned that the API should ideally be:

``def kanes_equations(bodies_and_particles, loads=None):`

But that would require a deprecation and change in the API due to the order change of the arguments. You will always have at least one particle or body but you may have no loads.

We could do this though, to stay backwards compatible:

``def KanesMethod.kanes_equations(loads=None, bodies_and_particles=None):`

This would allow the users to do any of these:

```python
kane.kanes_equations((body1, body2, particle1))
kane.kanes_equations((body1, ), (load1, load 2))
kane.kanes_equations((body1, body 2), loads=None)
kane.kanes_equations(bodies_and_particles=(body1, body2), loads=(load1, load2))
```

These would fail:

```python
kane.kanes_equations((body1, body2), loads=())
kane.kanes_equations((body1, body2), loads=[()])
```

BTW, I think we need a better name for "bodies_and_particles", something succint. But I guess a particle can be considered an infinitesimal sized body. I also prefer "loads" to forces, because the user can pass in forces or torques.

I think what I show above is the best we can do with the API and it will follow Python's Zen: "There should be one-- and preferably only one --obvious way to do it."